### PR TITLE
Fixed threadview error

### LINKED ISSUE #1970: [BUG] When accessing the threads of a channel or discussion
- Your Rocket.Chat Experimental app version: 4.6.0.9
- Your Rocket.Chat server version: 3.1.0-rc.6
- Device (or Simulator) you're running with: Android P Samsung Galaxy J6

As the Gif shows, when we enter thread view of channel or discussion, we get error message.
ezgif com-resize(1)

### PR DESCRIPTION
**Pull request by Prateek93a**
@RocketChat/ReactNative

Closes #1970 

As the Gif shows, the error is fixed. It was similar to #1444 .

ezgif com-resize(2)

### CODE REVIEW

File: app/views/ThreadMessagesView/index.js [hunk 1]
```diff
@@ -89,12 +89,13 @@ class ThreadMessagesView extends React.Component {
...
-	subscribeData = () => {
+	subscribeData = async() => {
 		try {
 			const db = database.active;
-			this.subObservable = db.collections
+			const subscription = await db.collections
 				.get('subscriptions')
-				.findAndObserve(this.rid);
+				.find(this.rid);
+			this.subObservable = subscription.observe();
```
Comment (diegolmello) [inline app/views/ThreadMessagesView/index.js:98]: `const observable` works?
I don't think we need `this.subObservable` elsewhere.

**Comment by diegolmello** [comment]
@Prateek93a It doesn't seem to fix the issue.

File: app/views/ThreadMessagesView/index.js (hunk 1 above)
Comment (Prateek93a) [reply app/views/ThreadMessagesView/index.js]: removed const declaration and this.subObservable.

**Comment by Prateek93a** [comment]
@diegolmello Yes Sir, the issue is not solved properly. Actually on rooms, which we have joined, the thread view does not show error now and shows all the threads and updates under changes properly, but on rooms we have not joined the thread view always shows empty screen even if there are threads in the room as the subscription record for that room is not found in the local db.

**Comment by diegolmello** [comment]
It's going to be fixed by #1997

**Pull request closed without merging**